src/VisualStudio/CSharp/Impl/Options/CSharpSettingsManagerOptionSerializer.cs

```diff
@@ -103,6 +103,18 @@ private bool ShouldIncludeOnOffOption(FieldInfo fieldInfo)
 
         protected override string SettingStorageRoot { get { return "TextEditor.CSharp.Specific."; } }
 
+        protected override string GetStorageKeyForOption(IOption option)
+        {
+            var name = option.Name;
+            if (option == ServiceFeatureOnOffOptions.ClosedFileDiagnostic)
+            {
+                // ClosedFileDiagnostics has been deprecated in favor of CSharpClosedFileDiagnostics
```

Comment:
I'd expect a comment here explaining the whole "why?" that this is trying to fix.